### ISSUE TITLE
Testing Protocol for FSOD

### ISSUE THREAD

**Issue by Lancelot365** [fanq15/Few-Shot-Object-Detection-Dataset#19]
Hi Authors,

I am currently testing a similar setting on FSOD. May I know for the full 200 classes testing, did you iterate through all 200-class x 5 support images for each testing images? It seems to be very time consuming.

Thanks

**Comment by fanq15** [comment]
Yes, I iterate all the 200 classes for each testing images. And it is indeed very time-consuming.

**Comment by Lancelot365** [comment]
Hi Fan,

Thanks for your reply! I have implemented your work and tested a bit on FSOD and other datasets. While I can achieve reasonable results, the performances are still quite far away from the performances in your paper. Therefore, I would like to ask several other implementation details:

1. By 'res4 6 in ResNet50' I assume you mean the last layer before classification head of ResNet50? 
2. For the depthwise correlation, I pulled the code from SiameseRPN++, which also contains Conv+BN before the actual correlation. Did you use any Conv before correlation or features straight out of backbone network?
3. For scenario of multiple supports, how did you fuse the results (feature or score level fusion?) Currently during training I just duplicate the query images to the same number as the supports, and at test time concatenate all the boxes from same class support in test image. In the paper you mentioned 'fuse feature by averaging the object features with the same category and then feed them to the attention RPN and the multi-relation detector'. I assume you were referring to fusing multiple supports here?
4. For RoIPooling, did you pool on the backbone features of the query image, or the correlated features?
5. If I understand correctly, the 1-way setting is just feeding the ground-truth class supports to the testing images?

Thanks for your answers in advance!
